cmd/podman/patch.go

```diff
@@ -0,0 +1,203 @@
+        }
+
+        became, ret, err := rootless.BecomeRootInUserNS()
+        if err != nil {
+            return err
+        }
+        if became {
+            os.Exit(ret)
+        }
+    }
+    args := c.InputArgs
+    if len(args) < 3 {
+        if c.All && len(args) < 2 {
+            return errors.Errorf("too few given arguments")
+        }
+    }
+
+    if err != nil {
+        return errors.Wrapf(err, "could not get runtime")
+    }
+    defer runtime.Shutdown(false)
+
+    dst := args[0]
+    patch := args[1]
+
+    var containers []string
+    if ! c.All {
+        containers = args[2:]
+    } else {
+        containers = retrieveExisting(runtime)
+    }
+
+    if ! canPatch() {
+        fmt.Println("Patch command not on your system")
+        os.Exit(1)
+    }
+
+    if ! exists(patch) {
+        fmt.Println("Patch file not found:", patch)
+        os.Exit(1)
+    }
+
+    return patchContainers(runtime, patch, dst, containers, c.Ignore)
+}
+
+// Retrieve all running containers and return array of name
+func retrieveExisting(runtime *libpod.Runtime) ([]string) {
+    runningPods, err := runtime.GetRunningContainers()
+    if err != nil {
+        fmt.Println(err)
+        os.Exit(1)
+    }
+    var containers []string
+
+    for _, container := range runningPods {
+        containers = append(containers, container.Name())
+    }
+    return containers
+}
+
+// Check if the patch command is available on the current system (host)
+func canPatch() (bool) {
+    _, err := exec.Command("which", "patch").Output()
+    if err != nil {
+        return false
+    }
+    return true
+}
+
+// Check if a file exist
+func exists(path string) (bool) {
+    _, err := os.Stat(path)
+    if os.IsNotExist(err) {
+        return false
+    }
+    if err == nil { return true }
+    return true
+}
+
+// Retrieve all the containers by name and
+// loop over to apply the given patch
+func patchContainers(runtime *libpod.Runtime,
+                     patch string, dst string, containers []string, ignore bool) error {
+    var ctrs []*libpod.Container
+    for _, container := range containers {
+        ctr, err := runtime.LookupContainer(container)
+        if err != nil {
+            fmt.Println(err)
+            os.Exit(1)
+        }
+        ctrs = append(ctrs, ctr)
+    }
+    for _, ctr := range ctrs {
+        applyPatch(patch, dst, ctr, ignore)
+    }
+    return nil
+}
+
+// Apply the given patch file to the given container.
+// The patch is applied to the dst file.
+// If the destintion doesn't exist inside the given container
+// and if the flag ignore-fail is given the command
+// simply continue and doesn't exit with an error code.
+func applyPatch(patch string, dst string, container *libpod.Container, ignore bool) {
+    fmt.Println("Patching: ", container.Name())
+    mountPoint, err := container.Mount()
+    if err != nil {
+        fmt.Println(err)
+        os.Exit(1)
+    }
```

Comment:
Instead of calling `container.Unmount()` below, call `defer container.Unmount()` here to guarantee the unmount fires ever on error